gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+
+*Important: The service may require that you verify ownership of the domain name that you are
+creating a zone for. Hence, we recommend that you do so beforehand. You can verify ownership of
+a domain name [here](https://www.google.com/webmasters/verification/home). Note that `gcloud-java`
+works with fully qualified domain names which must end with a period.*
+
+```java
+// Create a zone metadata
+String zoneName = "my_unique_zone"; // Change this zone name which is unique within your project
+String domainName = "someexampledomain.com."; // Change this to a domain which you own
+String description = "This is a gcloud-java-dns sample zone.";
+ZoneInfo zoneInfo = ZoneInfo.of(zoneName, domainName, description);
+
+// Create zone in Google Cloud DNS
+Zone zone = dns.create(zoneInfo);
+```
+
+You now have an empty zone hosted in Google Cloud DNS which is ready to be populated with DNS
+records for domain name `someexampledomain.com.` Upon creating the zone, the cloud service
+automatically assigned a set of DNS servers to this zone which will be hosting the records, and
+created the default required SOA and NS records for the domain inside this zone. You can read the
+list of assigned DNS servers using accessor method `zone.nameServers()` and for instance print them
+on the screen as follows. You will need to add `import com.google.gcloud.dns.DnsRecord` to your
+source code.
+
+```java
+// Print assigned name servers
+List<DnsRecord> records = zone.nameServers();
+for(DnsRecord record : records) {
+  System.out.println(record)
+}
+```
+
+You can now instruct your domain registrar to [update the domain name servers] (https://cloud.google.com/dns/update-name-servers)
+to these. As soon as this happens and the change propagates through cached values in DNS resolvers,
+all the DNS queries will be directed and answered by the Google Cloud DNS service.
+
+#### Creating DNS Records
+Now that we have an zone, we should add some DNS records. The DNS records held within zones are
+modified by so-called "change requests". In this example, we create and apply a change request to
+our zone that creates a DNS record of type A and points URL www.someexampledomain.com to
+IP address 12.13.14.15. Start by adding `import com.google.gcloud.dns.ChangeRequest` and proceed
+with:
+
+```java
+// Prepare a www.someexampledomain.com. type A record with ttl of 24 hours
+String ip = "12.13.14.15";
+DnsRecord toCreate = DnsRecord.builder("www.someexampledomain.com.", DnsRecord.Type.A)
+  .ttl(24, TimeUnit.HOURS)
+  .addRecord(ip)
+  .build();
+// Make a change
+ChangeRequest.Builder changeBuilder = ChangeRequest.builder().add(toCreate);
+// Build and apply the change request to our zone
+ChangeRequest changeRequest = zone.applyChangeRequest(changeBuilder.build());
+```
+
+The `addRecord` (and also `records`) function of `DnsRecord.Builder` accepts records in the form of
```

Comment:
I think you can remove the mentions of the `records` method.  Those mentions make this sentence a more complex, and as users can find that by reading more docs.